### PR TITLE
Cleanup the code and star using new EmbedLiteWindow object

### PR DESCRIPTION
**Pull request by tworaz**
This basically drops old Qt4 embed widgets, And adjusts QtMozEmbed layer to EmbedLite changes from https://github.com/tmeshkova/gecko-dev/pull/86. The code should work for bot sailfish-browser and qmlMozEmbedTestQt5. The QOpenGLWebPage part is also fully testable on desktop via small browser app I wrote: https://github.com/tworaz/qzilla/tree/gecko38

### CODE REVIEW

**Comment by tworaz** [comment]
This patchset depends on https://github.com/tmeshkova/gecko-dev/pull/86.

File: src/qmozview_p.cpp [hunk 1]
```diff
@@ -433,54 +418,31 @@ void QGraphicsMozViewPrivate::sendAsyncMessage(const QString &name, const QVaria
...
-            break;
+    mMozWindow = window;
+    if (mMozWindow) {
+        if (!mSize.isEmpty()) {
+            mMozWindow->setSize(mSize.toSize());
         }
-        mView->SetScreenRotation(rotation);
-        mOrientationDirty = false;
+        connect(mMozWindow.data(), &QMozWindow::compositorCreated,
+                this, &QMozViewPrivate::onCompositorCreated);
```
Comment (rainemak) [inline src/qmozview_p.cpp]: Could done like below. The ```onCompositorCreated``` slot is just a setter and not needed in the private API. This way we'd have a few lines less in the private class.

```
connect(mMozWindow.data(), &QMozWindow::compositorCreated, this, [=]() {
    mHasCompositor = true;
});
```

**Comment by rainemak** [comment]
The cleanup commit https://github.com/tworaz/qtmozembed/commit/a7c422397d8fdae3a9466a16921abdc3812645fc should be a separate PR. 

This PR is about taking new APIs in to use. Thus, I'd also create a separate PR for the renaming the private class (https://github.com/tworaz/qtmozembed/commit/1eb97e459a94962bc3c13ee5e2081ce405b60e83).

**Comment by rainemak** [comment]
Also commit https://github.com/tworaz/qtmozembed/commit/18422e8502143d297a84c493a2b2a7998f7d39c6 would be a natural separate PR.

**Comment by rojkov** [comment]
Tested and proved it to work with Gecko38. The comments should be addressed in separate PRs.

**Pull request merged**